jdaviz/configs/specviz2d/plugins/parsers.py

```diff
@@ -46,9 +46,15 @@ def spec2d_1d_parser(app, data_obj, data_label=None, show_in_viewer=True):
             flux = np.sum(data, header['DISPAXIS']-1)*u.Unit(header["BUNIT"])
         except ValueError:
             flux = u.Quantity(np.sum(data, header['DISPAXIS']-1))
-        step = (header["WAVEND"] - header["WAVSTART"])/len(flux)
-        spectral_axis = np.arange(header["WAVSTART"], header["WAVEND"],
-                                  step) * u.Unit("m")
+
+        if "WAVEND" in header and "WAVSTART" in header:
+            step = (header["WAVEND"] - header["WAVSTART"])/len(flux)
+            spectral_axis = np.arange(header["WAVSTART"], header["WAVEND"],
+                                      step) * u.Unit("m")
```

Comment:
Just use the unit instance directly.
```suggestion
                                      step) * u.m
```